src/fungal_effects.cpp

```diff
@@ -92,200 +92,148 @@ void fungal_effects::create_spores( const tripoint &p, Creature *origin )
+                    m.ter_set( p, t_fungus );
+                    gm.summon_mon( mon_fungal_blossom, p );
+                    if( gm.u.sees( p ) ) {
+                        add_msg( m_warning, _( "The young tree blooms forth into a fungal blossom!" ) );
+                    }
+                } else if( x_in_y( growth * 10, 400 ) ) {
+                    m.ter_set( p, t_marloss_tree );
+                }
+            } else {
                 m.ter_set( p, t_tree_fungal_young );
-                converted = true;
-            }
-        } else if( m.has_flag( "WALL", p ) && m.has_flag( "FLAMMABLE", p ) ) {
-            if( x_in_y( growth * 10, 5000 ) ) {
-                converted = true;
-                m.ter_set( p, t_fungus_wall );
             }
+            converted = true;
         }
-        // Furniture conversion
-        if( converted ) {
-            if( m.has_flag( "FLOWER", p ) ) {
-                m.furn_set( p, f_flower_fungal );
-            } else if( m.has_flag( "ORGANIC", p ) ) {
-                if( m.furn( p ).obj().movecost == -10 ) {
-                    m.furn_set( p, f_fungal_mass );
-                } else {
-                    m.furn_set( p, f_fungal_clump );
+    } else if( m.has_flag( "TREE", p ) ) {
+        if( one_in( 10 ) ) {
+            if( m.get_field_strength( p, fd_fungal_haze ) != 0 ) {
+                if( x_in_y( growth * 10, 100 ) ) {
+                    m.ter_set( p, t_fungus );
+                    gm.summon_mon( mon_fungal_blossom, p );
+                    if( gm.u.sees( p ) ) {
+                        add_msg( m_warning, _( "The tree blooms forth into a fungal blossom!" ) );
+                    }
+                } else if( x_in_y( growth * 10, 600 ) ) {
+                    m.ter_set( p, t_marloss_tree );
                 }
-            } else if( m.has_flag( "PLANT", p ) ) {
-                // Replace the (already existing) seed
+            } else {
+                m.ter_set( p, t_tree_fungal );
+            }
+            converted = true;
+        }
+    } else if( m.has_flag( "WALL", p ) && m.has_flag( "FLAMMABLE", p ) ) {
+        if( x_in_y( growth * 10, 5000 ) ) {
+            m.ter_set( p, t_fungus_wall );
+            converted = true;
+        }
+    }
+    // Furniture conversion
+    if( converted ) {
+        if( m.has_flag( "FLOWER", p ) ) {
+            m.furn_set( p, f_flower_fungal );
+        } else if( m.has_flag( "ORGANIC", p ) ) {
+            if( m.furn( p ).obj().movecost == -10 ) {
+                m.furn_set( p, f_fungal_mass );
+            } else {
+                m.furn_set( p, f_fungal_clump );
+            }
+        } else if( m.has_flag( "PLANT", p ) ) {
+            // Replace the (already existing) seed
+            if( !m.i_at( p ).empty() ) {
                 m.i_at( p )[0] = item( "fungal_seeds", calendar::turn );
+            } else {
+                add_msg( m_debug, "No seed item in the PLANT terrain at position %d,%d,%d", p.x, p.y, p.z );
```

Comment:
I was imagining an error in `debug.log`, rather than a debug message in the message log.  That has the advantage that it will trigger a test failure if it arises.

Reply:
Fixed